tests/integration/test_distributed_load_balancing/test.py

```diff
@@ -9,9 +9,21 @@
 
 cluster = ClickHouseCluster(__file__)
 
-n1 = cluster.add_instance("n1", main_configs=["configs/remote_servers.xml"])
-n2 = cluster.add_instance("n2", main_configs=["configs/remote_servers.xml"])
-n3 = cluster.add_instance("n3", main_configs=["configs/remote_servers.xml"])
+n1 = cluster.add_instance(
+    "n1",
+    main_configs=["configs/remote_servers.xml"],
+    user_configs=["configs/users.xml"],
+)
```

Comment:
AFAICS the problem is not the hedge requests, so unlikely this will help.

For example

```
$ grep '{6e71b635e2fa4ccc91033af7bbd9035c}' test_distributed_load_balancing/_instances_0/n*/logs/clickhouse-server.log 
test_distributed_load_balancing/_instances_0/n1/logs/clickhouse-server.log:2023.05.16 07:12:53.630960 [ 9 ] {6e71b635e2fa4ccc91033af7bbd9035c} <Debug> executeQuery: (from 172.16.3.1:56080) SELECT * FROM dist (stage: Complete)
test_distributed_load_balancing/_instances_0/n1/logs/clickhouse-server.log:2023.05.16 07:12:53.631170 [ 9 ] {6e71b635e2fa4ccc91033af7bbd9035c} <Trace> ContextAccess (default): Access granted: SELECT(key) ON default.dist
test_distributed_load_balancing/_instances_0/n1/logs/clickhouse-server.log:2023.05.16 07:12:53.631269 [ 9 ] {6e71b635e2fa4ccc91033af7bbd9035c} <Trace> ContextAccess (default): Access granted: SELECT(key) ON default.dist
test_distributed_load_balancing/_instances_0/n1/logs/clickhouse-server.log:2023.05.16 07:12:53.631312 [ 9 ] {6e71b635e2fa4ccc91033af7bbd9035c} <Trace> InterpreterSelectQuery: Complete -> Complete
test_distributed_load_balancing/_instances_0/n1/logs/clickhouse-server.log:2023.05.16 07:12:53.632076 [ 272 ] {6e71b635e2fa4ccc91033af7bbd9035c} <Debug> Connection (n2:9000): Sent data for 2 scalars, total 2 rows in 4.6548e-05 sec., 42356 rows/sec., 68.00 B (1.37 MiB/sec.), compressed 0.4594594594594595 times to 148.00 B (2.98 MiB/sec.)
test_distributed_load_balancing/_instances_0/n1/logs/clickhouse-server.log:2023.05.16 07:12:53.633366 [ 9 ] {6e71b635e2fa4ccc91033af7bbd9035c} <Debug> MemoryTracker: Peak memory usage (for query): 1.27 MiB.
test_distributed_load_balancing/_instances_0/n1/logs/clickhouse-server.log:2023.05.16 07:12:53.633378 [ 9 ] {6e71b635e2fa4ccc91033af7bbd9035c} <Debug> TCPHandler: Processed in 0.002664444 sec.
```

This query was sent to `n2`, while it should be `n1`, if the problem was in hedged requests we would see more `Sent data for` messages.

I think this should help #49921

Reply:
In some failures I saw that we tried to connect to `n1` first, then after 50 ms to `n2` and continued working with `n2` - after recent PR we connect to replicas async, so hedged requests also can make this test flaky